plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/openimplementation/OpenImplementationPresenter.java

```diff
@@ -96,161 +91,149 @@ public void show(final EditorPartPresenter editorPartPresenter) {
+    //    } else {
+    context
+        .getWorkspaceRoot()
+        .getFile(symbolInformation.getLocation().getUri())
+        .then(
+            file -> {
+              if (file.isPresent()) {
+                editorAgent.openEditor(
+                    file.get(),
+                    new OpenEditorCallbackImpl() {
+                      @Override
+                      public void onEditorOpened(EditorPartPresenter editor) {
+                        setCursor(symbolInformation.getLocation().getRange());
+                      }
+                    });
               }
             });
+    // }
+    Scheduler.get().scheduleDeferred(() -> activeEditor.setFocus());
   }
 
   private void showNoImplementations(
       NoImplementationWidget noImplementationWidget, TextEditor editorPartPresenter) {
     int offset = editorPartPresenter.getCursorOffset();
     PositionConverter.PixelCoordinates coordinates =
         editorPartPresenter.getPositionConverter().offsetToPixel(offset);
-    Type type = dtoFactory.createDto(Type.class);
-    type.setFlags(-1);
-    noImplementationWidget.addItem(type);
+    SymbolInformation symbolInformation = new SymbolInformation();
+    symbolInformation.setKind(null);
+    noImplementationWidget.addItem(symbolInformation);
     noImplementationWidget.show(coordinates.getX(), coordinates.getY());
   }
 
   private void openOneImplementation(
-      ImplementationsDescriptorDTO implementationsDescriptor,
+      ImplementationsDescriptor implementationsDescriptor,
       NoImplementationWidget implementationWidget,
       TextEditor editorPartPresenter) {
     int offset = editorPartPresenter.getCursorOffset();
     PositionConverter.PixelCoordinates coordinates =
         editorPartPresenter.getPositionConverter().offsetToPixel(offset);
-    for (Type type : implementationsDescriptor.getImplementations()) {
-      implementationWidget.addItem(type);
+    for (SymbolInformation symbolInformation : implementationsDescriptor.getImplementations()) {
+      implementationWidget.addItem(symbolInformation);
     }
     implementationWidget.show(coordinates.getX(), coordinates.getY());
     implementationWidget.asElement().getStyle().setWidth(600 + "px");
   }
 
-  private void setCursor(final Region region) {
+  private void setCursor(final Range region) {
     if (!(editorAgent.getActiveEditor() instanceof TextEditor)) {
       return;
     }
+
     Scheduler.get()
         .scheduleDeferred(
-            new Scheduler.ScheduledCommand() {
-              @Override
-              public void execute() {
-                TextEditor editor = (TextEditor) editorAgent.getActiveEditor();
-                editor.setFocus();
-                editor
-                    .getDocument()
-                    .setSelectedRange(
-                        LinearRange.createWithStart(region.getOffset()).andLength(0), true);
-              }
+            () -> {
+              ((TextEditor) editorAgent.getActiveEditor())
```

Comment:
This is bogus: the active editor may have changed in the meantime and the cast will crash.